### ISSUE TITLE
Federated Subscriptions: Could not connect to websocket endpoint

### ISSUE THREAD

**Issue by puchm** [mercurius-js/mercurius#479]
I've been trying to set up federated subscriptions. I followed the docs for subscriptions without a custom pubsub (for now). This is the error I got when trying to subscribe to the subscription using the playground:
```
{
  "error": "Could not connect to websocket endpoint ws://localhost:3000/graphql. Please check if the endpoint url is correct."
}
```

My gateway is on port 3000 so it is trying to connect to the gateway via the WS protocol even though the service with the actual subscription is on port 3002.

Do I need to do any setup within my gateway to enable subscriptions? This is the current setup:

```
const gateway = Fastify();

gateway.register(mercurius, {
    gateway: {
        services: [
            {
                name: 'user',
                url: 'http://localhost:3001/graphql'
            }, {
                name: 'message',
                url: 'http://localhost:3002/graphql'
            }
        ],
        pollingInterval: 2000 // refresh every 2 seconds
    },
    graphiql: 'playground'
})
```

I think in general it would be a good idea to add an article to the docs about federated subscriptions. I also didn't fully understand which service actually handles the websocket connections. Is it done by the gateway or by the individual services? This would of course affect whether users also need to expose the individual services.

**Comment by mcollina** [comment]
Did you read https://github.com/mercurius-js/mercurius/blob/master/examples/gateway-subscription.js?

**Comment by puchm** [comment]
I didn't but I looked at it now. After setting up the gateway properly I faced a new error:

`Subscription field must return Async Iterable. Received: undefined`

I was confused for a pretty long time but I have now also found the solution to that. The issue seems to occur when a subscription's return type is a scalar type (e.g. String). Then, the subscribe resolver is never executed and the error is returned. This should maybe be fixed or it should be added to the docs.

**Comment by mcollina** [comment]
Can you create a reproducible example of your problem? Adapting the example above to crash with that error would be of help.

**Comment by puchm** [comment]
Here is a fork - I just changed 2 lines. Launch the gateway-subscriptions example and try the following query on playground:

```
subscription {
  commentAdded(postId: "test")
}
```

The error is this one:

```
{
  "error": {
    "name": "FormatedError",
    "message": "Unknown error",
    "originalError": "Subscription field must return Async Iterable. Received: undefined."
  }
}
```

I don't know if subscriptions returning scalar types are allowed with GraphQL. If they are, the bug should be fixed. Otherwise there should be a more explicit error message, e.g. `Subscription field must not return a scalar. Received: string` or something like that.

**Comment by mcollina** [comment]
I think this is fixable, we should just understand where we are causing the problem.

I would love to get a PR for this as I do not have much time on my hands right now.

**Comment by simoneb** [comment]
This sounds like a duplicate of https://github.com/mercurius-js/mercurius/issues/370. It's probably not a good first issue though as a few people have already looked into it and the root cause doesn't seem to be obvious.

**Comment by simoneb** [comment]
@mcollina @puchm can we close this issue as a duplicate of #370?

**Comment by puchm** [comment]
Yes - Closed as duplicate of #370
